client/src/app/core/routing/menu-guard.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core'
+import { CanActivate, CanDeactivate } from '@angular/router'
+import { MenuService } from '@app/core/menu'
+import { ScreenService } from '@app/shared/misc/screen.service'
+
+@Injectable()
+export class MenuGuards {
+  public static guards: any = []
+
+  static open () {
+    MenuGuards.guards.push(OpenMenuGuard)
+    return OpenMenuGuard
+  }
+
+  static close () {
+    MenuGuards.guards.push(CloseMenuGuard)
+    return CloseMenuGuard
+  }
+}
+
+abstract class MenuGuard implements CanActivate, CanDeactivate<any> {
+  public display = true
+  constructor (protected menu: MenuService, protected screen: ScreenService, display: boolean) {
+    this.display = display
+  }
+  canActivate (): boolean {
+    // small screens already have the site-wide onResize from screenService
+    // > medium screens have enough space to fit the administrative menus
+    if (!this.screen.isInMobileView() && this.screen.isInMediumView()) {
+      this.menu.isMenuDisplayed = this.display
+    }
+    return true
+  }
+  public canDeactivate = this.canActivate
```

Comment:
Remove `public` and put attributes below `display = true`

Reply:
This is a function, so it should be placed below the function it copies.

Reply:
No: https://travis-ci.org/Chocobozzz/PeerTube/jobs/659521986

Reply:
That's news to me !